csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.shared.cs

```diff
@@ -766,43 +777,49 @@ public ulong ProfilingStartTimeNs
         #region private methods
 
         private void Init(string modelPath, SessionOptions options,
-                          PrePackedWeightsContainer prepackedWeightsContainer = null)
+                          PrePackedWeightsContainer prepackedWeightsContainer = null, IntPtr? envHandle = null)
```

Comment:
We confine native handles to the code that invokes Native API. Otherwise, we do not pass them around.